### DISCUSSION TITLE
Tell RenovateBot to upgrade another package in same MR

### DISCUSSION THREAD

**Discussion by jelhan** [renovatebot/renovate#25902] [Suggest an Idea]
### Tell us more.

Sometimes dependencies must be upgraded at the same time. E.g. if an NPM package upgrade requires a higher version of a peer dependency. It's often easy to spot how this can be solved. And often there is another PR from RenovateBot trying to upgrade the peer dependency but failing as well as the new version is not allowed by current peer dependency range. A scenario well known as upgrade hell.

Currently a developer needs to step in manually and upgrade both dependencies in the same PR. E.g. by creating another PR with a feature branch in which the developer merged both PRs created by RenovateBot.

This has several disadvantages:

1. Developer needs to checkout code locally, pull, merge, push and create PR.
2. PR created manually does not contain the nice summary of the upgrade we got used to from RenovateBot. In best case developer links the other PRs created by RenovateBot.

It would be great if we could tell RebovateBot to upgrade another dependency in the same PR in this case. Dependabot has the concept of commands, which could be triggered via comments on the PR. Would be great if RenovateBot supports something similar for this case. E.g. a "@RenovateBot merge #123", which tells RenovateBot to pull in the upgrade from PR #123 in this PR.

**Comment by secustor** [comment]
Use [`packageRules`](https://docs.renovatebot.com/configuration-options/#packagerules) with [`groupName`](https://docs.renovatebot.com/configuration-options/#groupname)

**Reply by jelhan** [reply]
`packageRules` works fine if you _always_ want to upgrade the packages together. This request is about cases, for which dependencies should be normally upgraded independently but for a specific upgrade need to be done together.

**Reply by viceice** [reply]
this can be done selectively, I'm doing grouping with peers for major updates only in some projects.

**Reply by jelhan** [reply]
But I hope it won't be needed in the future. I hope they learn that they should handle peer dependencies better with overlapping version ranges. I don't expect that I always need to update those packages at the same time. And for most updates it wasn't needed. It's an exception and therefore `packageRules` is not an answer.

I know `packageRules` and use them for packages published from a monorepo in lockstep. They are awesome for that case. But are not a solution if only for a specific version packages need to be upgraded at the same time.

One example is Prettier and the ESLint plugin for Prettier. In most cases you can upgrade `prettier`, `eslint` and `prettier-plugin-eslint` independently from each other. But for Prettier v3, ESLint v8 and eslint-plugin-prettier v5 everything needed to be done in one PR. This feature request is about easing upgrades for the rare cases like this.
